### PR TITLE
Delete Chrome temp cache after closing

### LINKED ISSUE #119061: Chrome doesn't delete temp directory, eating up all disk space
## Steps to Reproduce

Continuously start an application with `flutter run -d chrome`.

**Expected results:** I can start the app all day in Chrome
**Actual results:** My disk runs out of space and my Mac stops working. The temp directory is extremely full.

image (4)

<details>
  <summary>Logs</summary>

```
[!] Flutter (Channel cleanup-chrome-cache-dir, 3.7.0-24.0.pre.88, on macOS 13.0.1 22A400 darwin-arm64, locale en-DE)
    • Flutter version 3.7.0-24.0.pre.88 on channel cleanup-chrome-cache-dir at /Users/pascalwelsch/Projects/github/passsy/flutter
    ! Warning: `flutter` on your path resolves to /opt/homebrew/Caskroom/flutter/3.0.4/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/pascalwelsch/Projects/github/passsy/flutter. Consider adding /Users/pascalwelsch/Projects/github/passsy/flutter/bin to the front of your
      path.
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.18.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/pascalwelsch/Projects/github/passsy/flutter. Consider adding /Users/pascalwelsch/Projects/github/passsy/flutter/bin to the front of your path.
    ! Upstream repository git@github.com:passsy/flutter.git is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to git@github.com:passsy/flutter.git to dismiss this error.
    • Framework revision cd35fee284 (11 minutes ago), 2023-01-24 13:26:18 +0100
    • Engine revision 7d3233d26d
    • Dart version 3.0.0 (build 3.0.0-154.0.dev)
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/pascalwelsch/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • Java binary at: /Users/pascalwelsch/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2022.1)
    • Android Studio at /Users/pascalwelsch/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/pascalwelsch/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/pascalwelsch/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 71.3.6
    • Dart plugin version 223.8214.16

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.3)
    • IntelliJ at /Users/pascalwelsch/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-1/213.7172.25/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3)
    • IntelliJ at /Users/pascalwelsch/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.7571.182/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.1)
    • IntelliJ at /Users/pascalwelsch/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8214.52/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (3 available)
    • Pascal’s iPhone XR (mobile) • 00008020-000D0C34263B002E • ios            • iOS 15.6.1 19G82
    • macOS (desktop)             • macos                     • darwin-arm64   • macOS 13.0.1 22A400 darwin-arm64
    • Chrome (web)                • chrome                    • web-javascript • Google Chrome 109.0.5414.87

[✓] HTTP Host Availability
    • All required HTTP hosts are available

```

</details>

Related to https://github.com/flutter/flutter/issues/118557

### PR DESCRIPTION
**Pull request by passsy**
Fixes #119061

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by christopherfujino** [comment]
@christopherfujino please review

File: packages/flutter_tools/lib/src/web/chrome.dart [hunk 1]
```diff
@@ -222,6 +222,8 @@ class ChromiumLauncher {
     if (process != null && cacheDir != null) {
       unawaited(process.exitCode.whenComplete(() {
         _cacheUserSessionInformation(userDataDir, cacheDir);
+        // cleanup temp dir
+        userDataDir.deleteSync(recursive: true);
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/web/chrome.dart:226]: Can you catch FileSystemExceptions here, as its not really actionable?

```suggestion
        try {
          userDataDir.deleteSync(recursive: true);
        } on FileSystemException {
          // ignore
        }
```

**Review by christopherfujino — APPROVED** [review]
LGTM, thanks for the fix!

**Comment by christopherfujino** [comment]
You'll have to re-base upstream to make `ci.yaml validation` happy

**Review by andrewkolos — APPROVED** [review]
LGTM

**Pull request merged**